pkg/controllers/management/feature/feature_handler.go

```diff
@@ -40,9 +43,23 @@ func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
 	if obj.Name == features.TokenHashing.Name() {
 		return obj, h.refreshTokens()
 	}
+
+	if obj.Name == features.Harvester.Name() {
+		return obj, h.toggleHarvesterNodeDriver(obj.Name)
+	}
 	return obj, nil
 }
 
+func (h *handler) toggleHarvesterNodeDriver(harvester string) error {
+	lister := h.mgmt.Management.NodeDrivers("").Controller().Lister()
+	cli := h.mgmt.Management.NodeDrivers("")
+	m, _ := lister.Get("", harvester)
+	val := features.GetFeatureByName(harvester).Enabled()
+	m.Spec.Active = val
```

Comment:
According to https://github.com/rancher/rancher/issues/34886#issuecomment-931785510, we don't need to deactivate the node driver when the feature flag is turned off.

Reply:
@gitlawr updated. Many thanks for the heads-up.